pkg/operator2/configobservation/oauth/observe_tokenconfig.go

```diff
@@ -0,0 +1,89 @@
+package oauth
+
+import (
+	"k8s.io/klog"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	"github.com/openshift/cluster-authentication-operator/pkg/operator2/configobservation"
+)
+
+const defaultAccessTokenMaxAgeSeconds = float64(86400) // a day
```

Comment:
Given that the golant struct fields are int32 and the value in seconds is fixed point, is it desirable to use a floating point type?

Reply:
glad you asked, this was such a PITA to get right.

The config is really a json that gets deserialized into golang nested interfaces.

When we make this int32 and store as such, the information about it being int is obviously stored.

Eventually, the `configObserver`, which lives in library-go, compares the config it got originally with the one that it got as observed config.

This is where all the fun happens. because the json parser actually likes to think of numbers as of floats prior to them being integers. That means that if we make this integer, same jsons will actually be considered different in their go interpretation even if our comparison here tells you the numbers are equal. That results in hotloop trying to update the `observedConfig` field on same values.

I tried many things, played with the json parser flags, use `json.Number` as the default representation for numbers found in the jsons we have here but nothing except this really worked.

I did a terrible job with code comments about this, I see.